packages/model/src/-private/errors.ts

```diff
@@ -13,7 +12,7 @@ type ValidationError = {
 /**
   @module @ember-data/model
 */
-interface ArrayProxyWithCustomOverrides<T, M = T> extends Omit<ArrayProxy<T, M>, 'clear' | 'content'> {
+interface ArrayProxyWithCustomOverrides<T> extends Omit<ArrayProxy<T>, 'clear' | 'content'> {
```

Comment:
I believe we had a types override for this. The reason we do this is that the existing types for array proxy are incorrect: they assume the content type is the same as the materialized type, when in fact ArrayProxy typically transforms content by running an operation for each member of content on access.

Reply:
The above said, we *might* not need this anymore given we don't actually use ArrayProxy anywhere but Errors at this point. Since it's not throwing a bunch of tsc errors that we've silenced (from what I can tell) lets go with the "delete shit" approach for now, just wanted to add context.